mimic/model/maas_objects.py

```diff
@@ -600,6 +603,55 @@ def get_host_info(self, available_types, requested_types, entity_id, clock):
                            current_timestamp)))
                      for rtype in requested_types])
 
+    def list_connections(self):
+        """
+        Lists connections that this agent has.
+        """
+        return [AgentConnection(id=self.id + '_' + dc,
+                                guid=self.connection_guid,
+                                agent_id=self.id,
+                                endpoint=dc,
+                                datacenter=dc)
+                for dc in ('dfw1', 'ord1', 'lon3')]
+
+
+@attributes([Attribute('id', instance_of=text_type),
+             Attribute('guid', instance_of=text_type),
+             Attribute('agent_id', instance_of=text_type),
+             Attribute('endpoint', instance_of=text_type),
+             Attribute('datacenter', instance_of=text_type),
+             Attribute('process_version', instance_of=text_type, default_value='1.0.0-mimic'),
+             Attribute('bundle_version', instance_of=text_type, default_value='1.0.0-mimic'),
+             Attribute('agent_ip', instance_of=text_type, default_value='::1'),
+             Attribute('features', instance_of=list, default_factory=lambda: [
+                 {'version': '1.0.0-mimic',
+                  'name': 'upgrades'},
+                 {'version': '1.0.0-mimic',
+                  'name': 'confd'},
+                 {'version': '1.0.0-mimic',
+                  'name': 'health'}]),
+             Attribute('agent_port',
+                       instance_of=text_type,
+                       default_factory=lambda: text_type(random_port()))])
+class AgentConnection(object):
+    """
+    Models an agent connection, which can be queried from MaaS.
+    """
+    def to_json(self):
+        """
+        Serializes this AgentConnection to a JSON-encodable dict.
+        """
+        return {'id': self.id,
```

Comment:
One nice thing about `attrs` is that this functionality is built in: `attr.asdict(self)`:

https://attrs.readthedocs.org/en/stable/api.html?highlight=asdict#attr.asdict